syntax = "proto3";

package pb;

option go_package = "github.com/NordSecurity/nordvpn-linux/daemon/pb";

import "config/protocol.proto";
import "config/technology.proto";
import "config/group.proto";

enum ConnectionSource {
  UNKNOWN_SOURCE = 0;
  MANUAL = 1;
  AUTO = 2;
}

message ConnectionParameters {
  ConnectionSource source = 1;
  string country = 2;
  string city = 3;
  config.ServerGroup group = 4;
}

message StatusResponse {
  string state = 1;
  config.Technology technology = 2;
  config.Protocol protocol = 3;
  string ip = 4;
  string hostname = 5;
  string country = 6;
  string city = 7;
  uint64 download = 8;
  uint64 upload = 9;
  int64 uptime = 10;
  string name = 11;
  bool virtualLocation = 12;
  ConnectionParameters parameters = 13;
  bool postQuantum = 14;
}
